lib/commit.js

```diff
@@ -50,6 +51,135 @@ Commit.prototype.amend = function (
       });
 };
 
+/**
+ * Amend a commit with the given signature
+ * @async
+ * @param {String} update_ref
+ * @param {Signature} author
+ * @param {Signature} committer
+ * @param {String} message_encoding
+ * @param {String} message
+ * @param {Tree|Oid} tree
+ * @param {String} signature_field
+ * @param {Function} onSignature
+ * @return {Oid}
+*/
+Commit.prototype.amendWithSignature = function(
+  updateRef,
+  author,
+  committer,
+  message_encoding,
+  message,
+  tree,
+  signature_field,
+  onSignature
+) {
+    var repo = this.repo;
+    var parentOids = this.parents();
+    var _this = this;
+    var promises = [];
+
+    if (tree instanceof NodeGit.Oid) {
+      promises.push(repo.getTree(tree));
+    } else {
+      promises.push(Promise.resolve(tree));
+    }
+
+    parentOids.forEach(function (parentOid) {
+      promises.push(repo.getCommit(parentOid));
+    });
+
+    var treeObject;
+    var parents;
+    var commitContent;
+    var commitOid;
+    var commit;
+
+    var createCommitPromise = Promise.all(promises)
+      .then(function(results) {
+        treeObject = results[0];
+
+        parents = [];
+        for (var i = 0; i < parentOids.length; i++) {
```

Comment:
if you're going to use fp in this method, we should use fp here, as well.
`parents = fp.tail(results);`